sssom/util.py

```diff
@@ -146,14 +146,18 @@ def __str__(self) -> str:  # noqa:D105
     def clean_prefix_map(self) -> None:
         """Remove unused prefixes from the internal prefix map based on the internal dataframe."""
         prefixes_in_map = get_prefixes_used_in_table(self.df)
+        if self.metadata:
+            prefixes_in_metadata = get_prefixes_used_in_metadata(self.metadata)
         new_prefixes: PrefixMap = dict()
         missing_prefixes = []
         for prefix in prefixes_in_map:
```

Comment:
```suggestion
   all_prefixes = prefixes_in_metadata + prefixes_in_table
        for prefix in all_prefixes:
```

Reply:
This is just for illustration, you can just do it on your own branch

Reply:
I thought of doing this initially : `all_prefixes = prefixes_in_metadata + prefixes_in_table` but then checkout line 165:
```
self.prefix_map = new_prefixes
```
The reason I keep them separate is because that way I'll know which one belongs to the `prefix_map` and which one to the `metadata`. The differentiation is needed (I think).